src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -7302,11 +7341,8 @@ internal Task WriteSqlVariantValue(object value, int length, int offset, TdsPars
 
                 case TdsEnums.SQLUNIQUEID:
                     {
-                        System.Guid guid = (System.Guid)value;
-                        byte[] b = guid.ToByteArray();
-
-                        Debug.Assert((length == b.Length) && (length == 16), "Invalid length for guid type in com+ object");
-                        stateObj.WriteByteArray(b, length, 0);
+                        WriteGuid((Guid)value, stateObj);
+                        Debug.Assert(length == 16, "Invalid length for guid type in com+ object");
```

Comment:
You should probably invert these lines. Generally asserts come before the work they are preconditions for.